x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/AuthenticationFailureHandler.java

```diff
@@ -106,4 +109,11 @@ ElasticsearchSecurityException failedAuthentication(TransportMessage message, Au
      * @return ElasticsearchSecurityException with the appropriate headers and message
      */
     ElasticsearchSecurityException authenticationRequired(String action, ThreadContext context);
+
+    /**
+     * This method is called when failureResponseHeaders need to be set (at startup) or updated (if license state changes)
+     *
+     * @param failureResponseHeaders the Map of failure response heaers to be set
+     */
+    void setHeaders(Map<String, List<String>> failureResponseHeaders);
```

Comment:
I don't think we want to add this to the interface, because we actually only call it on the `DefaultAuthenticationFailureHandler`.

Instead we need to rework the code `Security.java` so that the `final` handler is the correct type.

Reply:
Just changed